### PR TITLE
[bots] Move labeler action to test-infra

### PR DESCRIPTION
**Pull request by clee2000**
Attempts to mimic the behavior of https://github.com/pytorch/pytorch/blob/89add71168faf48ea408829678897befcf0136b4/.github/workflows/labeler.yml

* Downloads the config based on a "labeler_config" key in .github/pytorch-probot.yml.  By nature of `context.config`, it looks in .github/ + configfile for the config file
* Uses minimatch (linked in https://github.com/actions/labeler/blob/83720bce8638d1b5b7b15ae76465522ca7ea2eec/README.md) to match the paths against globs
* Adds pull_request.synchronize to list of event types the labeler is run on

Other
* Adds check for ciflow label (no idea why this wasn't done before) 

Want to test this on malfet/deleteme before actually enabling on pytorch

### CODE REVIEW

File: torchci/lib/bot/utils.ts [hunk 1]
```diff
@@ -107,6 +107,35 @@ export class CachedIssueTracker extends CachedConfigTracker {
...
+      }
+    });
+  }
+
+  async loadLabelsConfig(context: Context, force = false): Promise<object> {
+    const key = repoKey(context);
+    if (!(key in this.repoLabels) || force) {
+      const config: any = await this.loadConfig(context, force);
+
+      if (config != null && "labeler_config" in config) {
```
Comment (huydhn) [inline torchci/lib/bot/utils.ts:129]: Just curious, I guess `labeler_config` on this context will come from https://github.com/pytorch/pytorch/blob/main/.github/labeler.yml.  Do you know how it happens?  Does GH does that for you when it sends the payload over to the bot?

File: torchci/lib/bot/autoLabelBot.ts [hunk 2]
```diff
@@ -324,56 +354,84 @@ function myBot(app: Probot): void {
...
-    const title = context.payload.pull_request.title;
-    const filesChangedRes = await context.octokit.paginate(
-      "GET /repos/{owner}/{repo}/pulls/{pull_number}/files",
-      {
-        owner,
-        repo,
-        pull_number: context.payload.pull_request.number,
-        per_page: 100,
+  app.on(
+    ["pull_request.opened", "pull_request.edited", "pull_request.synchronize"],
```
Comment (huydhn) [inline torchci/lib/bot/autoLabelBot.ts:358]: This looks slightly different than https://github.com/pytorch/pytorch/blob/main/.github/workflows/labeler.yml#L10 but having `pull_request.edited` makes sense to me as it covers updating base branch

**Comment by huydhn** [comment]
Is this done in the context of https://github.com/pytorch/test-infra/issues/4950 where we can get rid of labeler workflow?

There is this logic https://github.com/pytorch/pytorch/blob/main/.github/workflows/labeler.yml#L28 but I think it's obsolete and not used anymore.

The change LGTM but cc @izaitsevfb @kit1980 for the final say in case there are additional security concerns.  This should be safe I guess because there wouldn't be any workflow left (more secure)

File: torchci/lib/bot/utils.ts (hunk 1 above)
Comment (huydhn) [reply torchci/lib/bot/utils.ts:129]: Ah, spot your PR description, it's magic

> Downloads the config based on a "labeler_config" key in .github/pytorch-probot.yml. By nature of context.config, it looks in .github/ + configfile for the config file

**Comment by clee2000** [comment]
> Is this done in the context of https://github.com/pytorch/test-infra/issues/4950 where we can get rid of labeler workflow?

Yes 

> There is this logic https://github.com/pytorch/pytorch/blob/main/.github/workflows/labeler.yml#L28 but I think it's obsolete and not used anymore.

Just checked on that issue, it is closed and no longer used

**Review by huydhn — APPROVED** [review]

**Pull request merged**